linera-views/src/reentrant_collection_view.rs

```diff
@@ -506,7 +536,7 @@ where
                         let context = context.clone_with_base_key(key);
                         let view = W::new(context)?;
                         let view = Arc::new(RwLock::new(view));
-                        entry.insert(Update::Set(view));
+                        entry.insert(entry_wrapper(view));
```

Comment:
In the case of `try_load_entries` I believe we should return `None` instead of inserting `Unchanged(default_view)` for a key that doesn't existing in storage. (Otherwise this does change the list of keys and the hash.)